core/variant/array.cpp

```diff
@@ -298,6 +298,21 @@ Error Array::insert(int p_pos, const Variant &p_value) {
 	ERR_FAIL_COND_V_MSG(_p->read_only, ERR_LOCKED, "Array is in read-only state.");
 	Variant value = p_value;
 	ERR_FAIL_COND_V(!_p->typed.validate(value, "insert"), ERR_INVALID_PARAMETER);
+
+	if (p_pos < 0) {
+		// Relative offset from the end
+		p_pos = _p->array.size() + p_pos;
+	}
+
+	ERR_FAIL_INDEX_V_MSG(
+			p_pos,
+			_p->array.size(),
+			ERR_INVALID_PARAMETER,
+			vformat(
+					"The calculated index %s is out of bounds (the array has %s elements). Leaving the array untouched.",
+					p_pos,
+					_p->array.size()));
```

Comment:
```suggestion
	ERR_FAIL_INDEX_V(p_pos, _p->array.size(), ERR_INVALID_PARAMETER);
```
`ERR_FAIL_INDEX_V` already has a clear message